x/farming/keeper/plan.go

```diff
@@ -213,24 +250,66 @@ func (k Keeper) CreateRatioPlan(ctx sdk.Context, msg *types.MsgCreateRatioPlan,
 	return ratioPlan, nil
 }
 
-// TerminatePlan sends all remaining coins in the plan's farming pool to
-// the termination address and mark the plan as terminated.
+// TerminatePlan marks the plan as terminated.
+// It moves the plan under different store key, which is for terminated plans.
 func (k Keeper) TerminatePlan(ctx sdk.Context, plan types.PlanI) error {
+	_ = plan.SetTerminated(true)
+	k.SetPlan(ctx, plan)
+	if plan.GetType() == types.PlanTypePrivate {
+		k.DecrementNumPrivatePlans(ctx, 1)
+	}
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			types.EventTypePlanTerminated,
+			sdk.NewAttribute(types.AttributeKeyPlanId, strconv.FormatUint(plan.GetId(), 10)),
+		),
+	})
+
+	return nil
+}
+
+// RemovePlan removes a terminated plan and sends all remaining coins in the
+// farming pool address to the termination address.
+func (k Keeper) RemovePlan(ctx sdk.Context, creator sdk.AccAddress, planId uint64) error {
```

Comment:
How about refactoring this name to `RemovePrivatePlan` or expend this function also could cover public plan for `DeletePublicPlanProposal`

Reply:
The method name reflects the msg's name, `MsgRemovePlan`. Since we're using `MsgCreateFixedAmountPlan` and `MsgCreateRatioPlan`, not `MsgCreatePrivateFixedAmountPlan` or `MsgCreatePrivateRatioPlan`, I think it'd be better to use `MsgRemovePlan` as it keeps msg names consistent.
That being said, it's still possible to rename only this particular method to `RemovePrivatePlan` and call it in `msgServer.RemovePlan`. Let's discuss more about this on Slack.